lib/providers/compliance/s3.ts

```diff
@@ -0,0 +1,60 @@
+
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { default as Logger } from 'bunyan';
+
+import { checkDefined } from '../../preconditions/preconditions';
+import { FillerComplianceConfiguration, FillerComplianceConfigurationProvider } from '.';
+
+
+export class S3FillerComplianceConfigurationProvider implements FillerComplianceConfigurationProvider {
+  private log: Logger;
+  private configs: FillerComplianceConfiguration[];
+  private addrToEndpointsMap: Map<string, Set<string>>;
+
+  constructor(_log: Logger, private bucket: string, private key: string) {
+    this.configs = [];
+    this.log = _log.child({ quoter: 'S3FillerComplianceConfigurationProvider' });
+    this.addrToEndpointsMap = new Map<string, Set<string>>();
+  }
+
+  async getAddrToEndpointsMap(): Promise<Map<string, Set<string>>> {
+    if (this.configs.length === 0) {
```

Comment:
do we want to refetch configs on a timer to capture updates without a redeploy? Or maybe lambda container cycling does this for us

Reply:
Yeah I decided to not refetch on our own because 1) the config doesn't update as often, definitely not every 15 minutes 2) it's much larger than other configs we have so I'd rather we only read it once during lambda cold start